src/app/components/HoldRequest/HoldRequest.jsx

```diff
@@ -14,19 +15,42 @@ class HoldRequest extends React.Component {
   constructor(props) {
     super(props);
 
-    this.state = {
-      data: Store.getState(),
-      patron: PatronStore.getState(),
-    };
+    this.state = { patron: PatronStore.getState() };
+
+    // change all the components :(
     this.onChange = this.onChange.bind(this);
+    this.submitRequest = this.submitRequest.bind(this);
   }
 
   componentDidMount() {
     this.requireUser();
   }
 
   onChange() {
-    this.setState({ data: Store.getState() });
+    this.setState({ patron: PatronStore.getState() });
```

Comment:
We probably don't need to update `PatronStore`. There's no `Action` imported here either.